monkey/common/types/concurrency.py

```diff
@@ -22,28 +22,16 @@ def acquire(self, blocking: bool = ..., timeout: float = ...) -> bool:
     def release(self) -> None:
         ...
 
+
+class Lock(BasicLock, Protocol):
     def locked(self) -> bool:
         ...
 
 
-class RLock(Protocol):
+class RLock(BasicLock, Protocol):
     def __enter__(self, blocking: bool = ..., timeout: float = ...) -> bool:
```

Comment:
Huh, I _swear_ I tried this and it didn't like me overloading the context manager 🤷‍♂️